core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -192,13 +193,32 @@ private DefaultClientRequestContext(
         log.startRequest(requestStartTimeNanos, requestStartTimeMicros);
 
         if (responseCancellationScheduler == null) {
+            long responseTimeoutMillis = requestOptions.responseTimeoutMillis();
+            if (responseTimeoutMillis == -1) {
```

Comment:
How about using `< 0`? We always use `-1` to represent an unspecified value, but probably better being more careful when checking the value.